### PR TITLE
feat(aot): Universal 2.1.0-rc.1 - AoT support for Node/Browser

### PR DESCRIPTION
**Pull request by gdi2290**
```
npm install 
gulp pre-publish
cd to examples/express-aot
npm run ngc && npm run build
```

update:
I need to split this PR out so webpack, typescript, and angular updates will be done in another repo then rebased here. We're blocked on this PR until a PR is released in the angular repo or if we find a workaround

update:
- [x] update to 2.1.0

### CODE REVIEW

File: modules/platform-node/node-http.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ export class PreloadHttp extends Http {
 
 
 @Injectable()
-export class NodeConnection implements XHRConnection {
+export class NodeConnection /*implements XHRConnection*/ {
```
Comment (gdi2290) [inline modules/platform-node/node-http.ts]: ngc needs this

File: modules/platform-node/node-http.ts [hunk 2]
```diff
@@ -249,15 +249,15 @@ export class NodeConnection implements XHRConnection {
...
 
 @Injectable()
-export class NodeBackend extends XHRBackend {
+export class NodeBackend /*extends XHRBackend*/ {
   constructor(
     private baseResponseOptions: ResponseOptions,
-    _browserXHR: BrowserXhr,
-    _xsrfStrategy: XSRFStrategy,
+    // _browserXHR: BrowserXhr,
+    // _xsrfStrategy: XSRFStrategy,
```
Comment (gdi2290) [inline modules/platform-node/node-http.ts]: ngc needs these

**Comment by gdi2290** [comment]
ok I pushed up latest and I'm blocked by typescript and AoT someone can reproduce
```
npm install 
gulp pre-publish
cd to examples/express-aot
npm run ngc && npm run build
```

**Comment by gdi2290** [comment]
I need to split this PR out so webpack, typescript, and angular updates will be done in another repo then rebased here. We're blocked on this PR until a PR is released in the angular repo or if we find a workaround

**Comment by steve8708** [comment]
Just pushed an update to my [angular pr](https://github.com/angular/angular/pull/12158) that I've verified also resolves the runtime issues as well, so with that change the AOT example now builds and runs end to end for me 👍

**Comment by steve8708** [comment]
The PR has been reviewed and marked LGTM so should be merge anytime now

**Comment by steve8708** [comment]
blocker PR in angular is now merged 🎉

**Comment by steve8708** [comment]
just hooked a build of this branch up with the latest angular build to a fairly large project and the universal + aot integration works end to end like a charm 👍 

and my god it is FAST!!!  🎉 🎉 🎉

**Comment by steve8708** [comment]
I'm running prelimary load testing on the site I integrated universal + aot (compared to another branch on universal + jit) and I'm measuring anywhere from **4-10x faster render times** on the server, able to handle **2-10x more concurrency**, and is **4-5x more memory efficient** 

this is huge 🍾  🍾  🍾

**Comment by steve8708** [comment]
We will release something to production soon and I'll report back on the measured results in the wild

**Comment by MarkPieszak** [comment]
Wow really that big of gains? 😀😍😎

**Comment by elekzalan** [comment]
Hello. Any news here? I'm really excited to see those perf improvements. Keep up the good work!

**Comment by steve8708** [comment]
@elekzalan this has already been published with @angular/universal@2.1.0-rc.1

**Pull request closed without merging**